### PR TITLE
Fix rlc sample issue and enable more rlc tests

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/1577

### CODE REVIEW

File: packages/autorest.typescript/src/utils/valueHelpers.ts [hunk 1]
```diff
@@ -149,9 +157,11 @@ export function getParameterAssignment(exampleValue: ExampleValue, isRLCSample:
         }
         let propRetValue: string;
         if (propName.indexOf("/") > -1 || propName.match(/^\d/)) {
-          propRetValue = `"${propName}": ` + getParameterAssignment(property);
+          propRetValue =
+            `"${propName}": ` + getParameterAssignment(property, isRLCSample);
```
Comment (qiaozha) [inline packages/autorest.typescript/src/utils/valueHelpers.ts:161]: Have you tried to use your branch generate rlc for arm-compute and arm-network ?

File: packages/autorest.typescript/test/commands/smoke-test-list.ts [hunk 2]
```diff
@@ -230,6 +230,19 @@ export const readmes: SpecDefinition[] = [
       AutorestParams.GenerateSamples
     ],
     buildTag: "ci_rlc"
+  },
+  {
+    path: joinPath(
+      `${__dirname}`,
+      "..",
+      "./smoke/swagger/anomaly-detector-rest.md"
```
Comment (qiaozha) [inline packages/autorest.typescript/test/commands/smoke-test-list.ts:238]: why add this one into smoke-test list ?

File: packages/autorest.typescript/src/utils/valueHelpers.ts (hunk 1 above)
Comment (MaryGao) [reply packages/autorest.typescript/src/utils/valueHelpers.ts:161]: Yes but we still have issues when compilation.

For arm-network
```
samples-dev/applicationGatewayWafDynamicManifestsDefaultGetSample.ts:26:24 - error TS2322: Type '"2021-05-01"' is not assignable to type '"2022-05-01"'.

26     queryParameters: { "api-version": "2021-05-01" }
                          ~~~~~~~~~~~~~

  src/parameters.ts:465:3
    465   "api-version": "2022-05-01";
          ~~~~~~~~~~~~~
    The expected type comes from property 'api-version' which is declared here on type 'ApplicationGatewayWafDynamicManifestsDefaultGetQueryParamProperties & Record<string, unknown>'

samples-dev/applicationGatewayWafDynamicManifestsGetSample.ts:27:24 - error TS2322: Type '"2021-05-01"' is not assignable to type '"2022-05-01"'.

27     queryParameters: { "api-version": "2021-05-01" }
```

File: packages/autorest.typescript/test/commands/smoke-test-list.ts (hunk 2 above)
Comment (MaryGao) [reply packages/autorest.typescript/test/commands/smoke-test-list.ts:238]: I want to enable more sample generation testing for rlc

**Review by qiaozha — APPROVED** [review]

**Pull request merged**